### PR TITLE
Basic stack

### PR DESCRIPTION
**Pull request by DiegoMC9**
solves #22 creating a simple 16 bit, 16 entries long stack

### CODE REVIEW

File: src/stack.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+typedef struct Stack
+{
+    int maxsize;
+    int top;
+    uint16_t *elements;
+} Stack;
+
+// creates a new stack
+Stack* newStack(int size);
```
Comment (samgh96) [inline src/stack.h:11]: I'd prefer "create_stack" (snake_case in C is usually the way to go)

File: src/stack.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// frees memory allocated by the stack
+void freeStack(Stack* st);
```
Comment (samgh96) [inline src/stack.h:13]: Same here, use "delete_stack" or "free_stack" instead of the camelCase option

File: src/stack.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        st->elements[++st->top] = dir;
+    }
+}
+int pop(Stack *st)
+{
+    if (!isEmpty(st))
+    {
+        return st->elements[st->top--];
+    }
+    return -1;
```
Comment (samgh96) [inline src/stack.c:30]: Maybe not the best approach as -1 is a valid value in memory (0xFFFF), we need some kind of exit signal there when an underflow happens. 

Leave it like that, I'll fill an issue and we will think about it later.

**Review by samgh96 — COMMENTED** [review]
Be nice with me and rename those two functions pl0x <3

**Review by DavidMazarro — APPROVED** [review]
LGTM

File: src/stack.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    return st->top == -1;
+}
+int isFull(Stack *st)
+{
+    return st->top == st->maxsize;
+}
+int size(Stack *st)
+{
+    return st->top + 1;
+}
```
Comment (ignaciobll) [inline src/stack.c:47]: Hey! It's me! Commenting on merged PRs.

Take a look at your format, maybe at a project with multiple people its interesting to agree on a code convention. It's really easy in fact when you can use an autoformatter. You could take a look at https://clang.llvm.org/docs/ClangFormat.html

It's really easy to integrate with:
- [Emacs](https://github.com/lassik/emacs-format-all-the-code)
- [Vim](https://github.com/rhysd/vim-clang-format)
- [Visual Studio](https://marketplace.visualstudio.com/items?itemName=LLVMExtensions.ClangFormat)

Unfortunately, I couldn't found a link to any Nano configuration :disappointed:

**Pull request merged**